main.go

```diff
@@ -73,85 +73,94 @@ func main() {
 		klog.Fatalf("Error building submariner clientset: %s", err.Error())
 	}
 
-	kubeInformerFactory := kubeInformers.NewSharedInformerFactoryWithOptions(kubeClient, time.Second*30, kubeInformers.WithNamespace(ss.Namespace))
-	submarinerInformerFactory := submarinerInformers.NewSharedInformerFactoryWithOptions(submarinerClient, time.Second*30, submarinerInformers.WithNamespace(ss.Namespace))
-	
+	kubeInformerFactory := kubeInformers.NewSharedInformerFactoryWithOptions(kubeClient, time.Second*30,
+		kubeInformers.WithNamespace(submSpec.Namespace))
+	submarinerInformerFactory := submarinerInformers.NewSharedInformerFactoryWithOptions(submarinerClient, time.Second*30,
+		submarinerInformers.WithNamespace(submSpec.Namespace))
+
 	start := func(context.Context) {
-		var ce cableengine.CableEngine
-		localCluster, err := util.GetLocalCluster(ss)
+		localCluster, err := util.GetLocalCluster(submSpec)
 		if err != nil {
-			klog.Fatalf("Fatal error occurred while retrieving local cluster %v", localCluster)
+			klog.Fatalf("Fatal error occurred while retrieving local cluster from %v: %v", submSpec, err)
 		}
-		localEndpoint, err := util.GetLocalEndpoint(ss.ClusterId, "ipsec", nil, ss.NatEnabled, append(ss.ServiceCidr, ss.ClusterCidr...))
+		localEndpoint, err := util.GetLocalEndpoint(submSpec.ClusterId, "ipsec", nil, submSpec.NatEnabled,
+			append(submSpec.ServiceCidr, submSpec.ClusterCidr...))
 		if err != nil {
-			klog.Fatalf("Fatal error occurred while retrieving local endpoint %v", localCluster)
+			klog.Fatalf("Fatal error occurred while retrieving local endpoint from %v: %v", submSpec, err)
```

Comment:
Would not be better a %#v for submSpec?

Reply:
yeah it seems so - I wasn't aware of # - still learning Go ...

Reply:
https://golang.org/pkg/fmt/ shows there are a number of interesting formatting directives in Go ;-).

Reply:
yup - studying that ...